src/map/skills/weapon/bash.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef BASH_SKILL_HPP
+#define BASH_SKILL_HPP
+
+#include "battle_skill_categories.hpp"
+#include "battle.hpp"
+#include "pc.hpp"
+
+/**
+ * BashSkill - Implementation of SM_BASH and MS_BASH
+ * A basic melee attack skill that deals increased damage and has stun effect
+ */
+class BashSkill : public WeaponSkill {
+public:
+    // Core identification
+    uint16 get_skill_id() const override { 
```

Comment:
All skills have a skill id, so make the skill id a member of BattleSkill, add it to the constructor of all the base skills.
something like https://github.com/vstumpf/rathena/compare/refactor/cmake...vstumpf:rathena:refactor/skills#diff-d259011f9c18d10972c5036f69f7aee74575862f7ce3123ad3ea7527b759d711R11
https://github.com/vstumpf/rathena/compare/refactor/cmake...vstumpf:rathena:refactor/skills#diff-590e433f1e9ec7966ddb6a25dc976716c8e3c19261e4b517c2c657c59f8848ceR24